packages/host/app/components/operator-mode/card-schema-editor.gts

```diff
@@ -0,0 +1,163 @@
+  };
+}
+
+export default class CardSchemaEditor extends Component<Signature> {
+  <template>
+    <style>
+      .schema-editor-container {
+        margin-top: var(--boxel-sp);
+      }
+
+      .schema {
+        display: grid;
+        gap: var(--boxel-sp);
+        padding: var(--boxel-sp);
+      }
+
+      .pill {
+        border: 1px solid gray;
+        display: inline-block;
+        padding: var(--boxel-sp-xxxs) var(--boxel-sp-xs);
+        border-radius: 8px;
+        background-color: white;
+        font-weight: 600;
+      }
+
+      .realm-icon {
+        margin-right: var(--boxel-sp-xxxs);
+      }
+
+      .card-field {
+        background-color: white;
+      }
+
+      .card-field {
+        margin-bottom: var(--boxel-sp-xs);
+        padding: var(--boxel-sp);
+        border-radius: var(--boxel-border-radius);
+        display: flex;
+      }
+
+      .card-fields {
+        margin-top: var(--boxel-sp);
+      }
+
+      .left {
+        display: flex;
+        margin-top: auto;
+        margin-bottom: auto;
+        flex-direction: column;
+      }
+
+      .right {
+        margin-left: auto;
+        margin-top: auto;
+        margin-bottom: auto;
+      }
+
+      .field-name {
+        font-size: var(--boxel-font-size);
+      }
+
+      .field-type {
+        color: #949494;
+      }
+    </style>
+
+    <div
+      class='schema-editor-container'
+      data-test-card-schema={{@cardType.displayName}}
+    >
+      <div class='pill'>
+        <span class='realm-icon'>
+          🟦
+        </span>
+        {{@cardType.displayName}}
+      </div>
+
+      <div class='card-fields'>
+        {{#each @cardType.fields as |field|}}
+          {{#if (this.isOwnField field.name)}}
+            <div class='card-field' data-test-field-name={{field.name}}>
+              <div class='left'>
+                <div class='field-name'>
+                  {{field.name}}
+                </div>
+                <div class='field-type'>
+                  {{field.type}}
+                </div>
+              </div>
+              <div class='right'>
+
+                <div class='pill'>
+                  <span class='realm-icon'>
+                    🟪
+                  </span>
+                  {{#let
+                    (capitalize (this.cleanupCardType (cardId field.card)))
+                    as |cardType|
+                  }}
+                    <span data-test-card-type={{cardType}}>{{cardType}}</span>
+                  {{/let}}
+                </div>
+              </div>
+            </div>
+          {{/if}}
+        {{/each}}
+      </div>
+    </div>
+  </template>
+
+  @service declare loaderService: LoaderService;
+  @service declare cardService: CardService;
+
+  cleanupCardType(value: string) {
+    let path = new URL(value).pathname; // Examples of pathname: "/base/string/default", "/drafts/pet/Pet"
+
+    if (path.endsWith('/default')) {
+      path = path.slice(0, -'/default'.length);
```

Comment:
I am not sure if this is the correct approach to get the type. So for example when the path is "base/string/default", and we want to show "String" in the schema editor, I am removing the "default" part.

<img width="565" alt="image" src="https://github.com/cardstack/boxel/assets/273660/c0ade924-5e6e-4669-9f96-93943b2cd4fd">

Reply:
probably for the base fields we should have a static `displayName` property like cards can have?

Reply:
I like that idea

Reply:
Ok, I added `displayName` to base cards.